glide-core/src/client/value_conversion.rs

```diff
@@ -367,10 +376,87 @@ pub(crate) fn expected_type_for_cmd(cmd: &Cmd) -> Option<ExpectedReturnType> {
             }
         }
         b"LOLWUT" => Some(ExpectedReturnType::Lolwut),
+        b"XREAD" => Some(ExpectedReturnType::XreadReturnType),
         _ => None,
     }
 }
 
+fn convert_to_xread_map(value: Value) -> RedisResult<Value> {
+    // dbg!(value.clone());
+    match value {
+        Value::Array(array) => {
+            let mut map = Vec::new();
+            let mut iterator = array.into_iter();
+            while let Some(key) = iterator.next() {
```

Comment:
re-use `convert_array_to_map`, don't duplicate code

Reply:
If line 414 would be
```rs
                map.push((
                    convert_to_expected_type(inner_key, Some(ExpectedReturnType::BulkString))?,
                    convert_to_expected_type(inner_value, Some(ExpectedReturnType::XreadReturnType))?,
                ));
```
that will duplicate that function, I guess

Reply:
https://github.com/Bit-Quill/glide-for-redis/blob/7fe34bc54f4e509b585b5c6556d5e6de9a4eb841/glide-core/src/client/value_conversion.rs#L248-L273